ArcGISRuntimeSDKQt_QMLSamples/Routing/DisplayRouteLayer/DisplayRouteLayer.qml

```diff
@@ -0,0 +1,156 @@
+// [WriteFile Name=DisplayRouteLayer, Category=Routing]
+// [Legal]
+// Copyright 2022 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.12
+import Esri.ArcGISRuntime 100.15
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.11
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    property var featureCollection: null
+    property var featureCollectionLayer: null
+    property string directions: ""
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Component.onCompleted: {
+            // Set and keep the focus on MapView to enable keyboard navigation
+            forceActiveFocus();
+        }
+
+        PortalItem {
+            id: portalItem
+            itemId: "0e3c8e86b4544274b45ecb61c9f41336"
+            Component.onCompleted: load();
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded) {
+                    return;
+                }
+
+                // if the portal item is a feature collection, add the feature collection to the map's operational layers
+                if (type === Enums.PortalItemTypeFeatureCollection) {
+                    featureCollection = ArcGISRuntimeEnvironment.createObject("FeatureCollection", {item: portalItem});
+                    featureCollectionLayer = ArcGISRuntimeEnvironment.createObject("FeatureCollectionLayer", {featureCollection: featureCollection});
```

Comment:
`queryFeatures()` can be invoked here with the following, and then doesn't need to be called every time the popup is opened.
If this is done, then `featureCollection` can be `const` and doesn't need to be global. Nor would `featureCollectionLayer` need to be global.
```js
featureCollection.onLoadStatusChanged.connect(() => {
  if (featureCollection.loadStatus === Enums.LoadStatusLoaded) {
      const tables = featureCollection.tables;
      featureCollection.tables.forEach((table) => {
         if (table.displayName === "DirectionPoints")
         {
             if (table.loadStatus === Enums.LoadStatusLoaded)
             {
                 queryFeatures(table);
             }
         }
     });
   }
});
  ```

Reply:
Nice cleanup suggestion. Although I believe `featureCollection` still needs to be global to check the load status and enable/disable the Directions button.